server/game/cards/01_SOR/units/GideonHaskRuthlessLoyalist.ts

```diff
@@ -0,0 +1,28 @@
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { RelativePlayer, WildcardCardType } from '../../../core/Constants';
+import AbilityHelper from '../../../AbilityHelper';
+
+export default class GideonHaskRuthlessLoyalist extends NonLeaderUnitCard {
+    protected override getImplementationId () {
+        return {
+            id: '1664771721',
+            internalName: 'gideon-hask#ruthless-loyalist'
+        };
+    }
+
+    public override setupCardAbilities () {
+        this.addTriggeredAbility({
+            title: 'Give an Experience token to a friendly unit',
+            when: {
+                onCardDefeated: (event, context) => event.card.isUnit() && event.card.controller !== context.source.controller
+            },
+            targetResolver: {
+                controller: RelativePlayer.Self,
+                cardTypeFilter: WildcardCardType.Unit,
```

Comment:
nit: this type filter is technically not necessary, since `giveExperience()` can only target units there's some logic that will filter everything else out behind the scenes